searx/engines/public_domain_image_archive.py

```diff
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""Public domain image archive, based on the unsplash engine
+
+"""
+
+from urllib.parse import urlencode, urlparse, urlunparse, parse_qsl
+from json import loads
+
+THUMBNAIL_SUFFIX = "?fit=max&h=360&w=360"
+"""
+Example thumbnail urls (from requests & html):
+- https://the-public-domain-review.imgix.net/shop/nov-2023-prints-00043.jpg?fit=max&h=360&w=360
+- https://the-public-domain-review.imgix.net/collections/the-history-of-four-footed-beasts-and-serpents-1658/8616383182_5740fa7851_o.jpg?fit=max&h=360&w=360
+
+Example full image urls (from html)
+- https://the-public-domain-review.imgix.net/shop/nov-2023-prints-00043.jpg?fit=clip&w=970&h=800&auto=format,compress
+- https://the-public-domain-review.imgix.net/collections/the-history-of-four-footed-beasts-and-serpents-1658/8616383182_5740fa7851_o.jpg?fit=clip&w=310&h=800&auto=format,compress
+
+The thumbnail url from the request will be cleaned for the full image link
+The cleaned thumbnail url will have THUMBNAIL_SUFFIX added to them, based on the original thumbnail parameters
+"""
+
+# about
+about = {
+    "website": 'https://pdimagearchive.org',
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+
+base_url = 'https://oqi2j6v4iz-dsn.algolia.net/'
+search_url = (
+    base_url
+    + '1/indexes/*/queries?x-algolia-api-key=153d2a10ce67a0be5484de130a132050&x-algolia-application-id=OQI2J6V4IZ'
```

Comment:
That's basically a hardcoded API key, we should probably make that configurable, since that key probably differs per user?

Reply:
I've only confirmed using private windows, but I think the x-algolia-api-key might actually just be hardcoded like that in the original website. I can make it configurable either way, but would you mind checking if going to gives the same API key for you when capturing the request when searching at https://pdimagearchive.org/search/?q=dog?
![A screenshot of the Firefox dev console. The network tab shows the selected queries request, highlighting the x-algolia-api-key in the parameters below](https://github.com/user-attachments/assets/28bea13f-ff65-4267-b9c1-a5fb69f5a668)

Reply:
You're right, same API key for me, that's funny.

We can't be sure if it'll always stay the same or change at some point, so putting it into an `api_key` variable won't hurt :)